liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/BasicSupport.java

```diff
@@ -410,6 +410,33 @@ protected void installLicense() throws MojoExecutionException, IOException {
         }
     }
     
+    // delete application file from server apps or dropins directory, including loose application
+    protected void deleteApplication(File artifact) throws IOException {
+        deleteApplication(new File(serverDirectory, "apps"), artifact.getName());
+        if (artifact.getName().endsWith(".xml")) {
+            deleteApplication(new File(serverDirectory, "apps"), artifact.getName().substring(0, artifact.getName().length() - 4));
+        } else {
+            deleteApplication(new File(serverDirectory, "apps"), artifact.getName() + ".xml");
+        }
+        
+        deleteApplication(new File(serverDirectory, "dropins"), artifact.getName());
+        if (artifact.getName().endsWith(".xml")) {
```

Comment:
Since the logic is the same on these if/else statements you could collapse it into one statement with two calls or pass "app" and "dropins" as params and call the function twice in InstallApp